src/main/java/com/nike/cerberus/cache/DatabaseCache.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2019 Nike, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.nike.cerberus.cache;
+
+import com.codahale.metrics.Counter;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Injector;
+import com.nike.cerberus.server.config.guice.StaticInjector;
+import com.nike.cerberus.service.MetricsService;
+import com.typesafe.config.Config;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.ibatis.builder.InitializingObject;
+import org.apache.ibatis.cache.Cache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReadWriteLock;
+
+import static com.github.benmanes.caffeine.cache.Caffeine.newBuilder;
+import static java.util.Optional.ofNullable;
+
+/**
+ * This is a custom MyBatis Cache, that allows use to do the following
+ * 1. Report cache statistics via Dropwizard
+ * 2. Expire items automatically after some TTL from when they were cached. (To avoid needing to deal with distributed cache busting, this basically makes cached data eventually consistent up to the defined TTL)
+ * 3. Only cache items after it has been proven via repeat reads that they should be cached. (To avoid unnecessary eventual consistency in the dashboard, only make the items under heavy reads eventually consistent)
+ * See cms.conf for all the configuration settings.
+ */
+public class DatabaseCache implements Cache, InitializingObject {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+  private MetricsService metricsService;
+  private Integer repeatReadThreshold;
```

Comment:
Why is this not the primitive int?

Reply:
I make it an Integer because it is not being set in a constructor but rather the initialize method, that MyBatis calls internally. I would rather have it be null than 0 unintentionally.